tasks/eva_2357/populate_ids.py

```diff
@@ -0,0 +1,196 @@
+        alt = variant_query_result['alt']
+        id_variant_warehouse = variant_query_result['_id']
+        try:
+            genbank_chr, was_already_genbank = get_genbank(contig_synonym_dictionaries, variant_query_result['chr'])
+        except Exception as e:
+            logging.info(f"{e} in variant {variant_query_result['chr']}_{start}_{ref}_{alt}")
+            continue
+        sve_hash = get_SHA1(f"{assembly}_{study}_{genbank_chr}_{start}_{ref}_{alt}")
+        hash_to_variant_id[sve_hash] = id_variant_warehouse
+    return hash_to_variant_id
+
+
+def get_db_name_and_assembly_accession(databases):
+    db_assembly = {}
+    if databases:
+        with open(databases, 'r') as db_file:
+            for line in db_file:
+                db, assembly, asm_report = line.split(',')
+                db_assembly[db] = {"assembly": assembly, "asm_report": asm_report}
+        return db_assembly
+
+
+def populate_ids(private_config_xml_file, databases, profile='production', mongo_accession_db='eva_accession_sharded'):
+    db_assembly = get_db_name_and_assembly_accession(databases)
+    for db_name, info in db_assembly.items():
+        assembly = info['assembly']
+        asm_report = info['asm_report']
+        logging.info(f"Processing database {db_name} (assembly {assembly})")
+
+        contig_synonym_dictionaries = load_synonyms_for_assembly(assembly, asm_report)
+
+        with pymongo.MongoClient(get_mongo_uri_for_eva_profile(profile, private_config_xml_file)) as mongo_handle:
+            variants_collection = mongo_handle[db_name]["variants_2_0"]
+            logging.info(f"Querying variants from variant warehouse, database {db_name}")
+            variants_cursor = get_variants_from_variant_warehouse(variants_collection)
+            hash_to_variant_id = {}
+            update_statements = []
+            try:
+                for variant_query_result in variants_cursor:
+                    # One variant in the variant warehouse can include more than one study, which means there can be
+                    # more than one submitted variant per variant warehouse variant
+                    get_ids(assembly, contig_synonym_dictionaries, variant_query_result, hash_to_variant_id)
+                sve_hashes = hash_to_variant_id.keys()
+                logging.info(f"Querying accessioning database for assembly {assembly}")
+                hash_to_accession_info = get_from_accessioning_db(mongo_handle, mongo_accession_db, sve_hashes)
+                update_statements.extend(generate_update_statement(hash_to_variant_id, hash_to_accession_info))
+            except Exception as e:
+                print(traceback.format_exc())
+                raise e
+            finally:
+                variants_cursor.close()
+        logging.info(f"Updating database {db_name} variant warehouse with ss and rs ids")
+        result_update = variants_collection.bulk_write(requests=update_statements, ordered=False)
```

Comment:
Could you refer to [this documentation](https://docs.mongodb.com/v4.0/reference/method/db.collection.bulkWrite/#bulk-write-with-write-concern) to bulk write with a majority write concern and a generous timeout so that we don't get cursor timeout errors?